pkg/image/prune/prune.go

```diff
@@ -1002,13 +1011,14 @@ func (p *pruner) Prune(
 		return err
 	}
 
-	prunableComponents := getPrunableComponents(p.g, prunableImageIDs)
-
 	var errs []error
 
-	errs = append(errs, pruneImageComponents(p.g, p.registryClient, p.registryURL, prunableComponents, layerLinkPruner)...)
-	errs = append(errs, pruneBlobs(p.g, p.registryClient, p.registryURL, prunableComponents, blobPruner)...)
-	errs = append(errs, pruneManifests(p.g, p.registryClient, p.registryURL, prunableImageNodes, manifestPruner)...)
+	if p.algorithm.pruneRegistry {
+		prunableComponents := getPrunableComponents(p.g, prunableImageIDs)
+		errs = append(errs, pruneImageComponents(p.g, p.registryClient, p.registryURL, prunableComponents, layerLinkPruner)...)
+		errs = append(errs, pruneBlobs(p.g, p.registryClient, p.registryURL, prunableComponents, blobPruner)...)
+		errs = append(errs, pruneManifests(p.g, p.registryClient, p.registryURL, prunableImageNodes, manifestPruner)...)
+	}
 
 	if len(errs) > 0 {
```

Comment:
@gabemontero how about this condition ? please move it too inside `if p.algorithm.pruneRegistry`.

Reply:
yep part of next push